kubeflow/argo/prototypes/argo.jsonnet

```diff
@@ -3,12 +3,13 @@
 // @description Deploy Argo workflow engine
 // @shortDescription Argo workflow engine
 // @param name string Name to give to the component
-// @optionalParam namespace string default Namespace to use for the components.
-
-// TODO(https://github.com/ksonnet/ksonnet/issues/222): We have to add namespace as an explicit parameter
-// because ksonnet doesn't support inheriting it from the environment yet.
 
 local k = import "k.libsonnet";
 local argo = import "kubeflow/argo/argo.libsonnet";
 
-std.prune(k.core.v1.list.new(argo.parts(params.namespace).all))
+// updatedParams includes the namespace from env by default.
+// We can override namespace in params if needed
+local updatedParams = env + params;
```

Comment:
env is the params from environment?
Aren't we already setting per environment param to override default?

Reply:
`env` is not `params`. `params` are defined per component in `component/params.libsonnet` and can be overridden per-environment in `environments/<env>/params.libsonnet`

`env` is the ksonnet environment definition. It is the value you get when you execute `ks env list`

For example
```
% ks env list
NAME    KUBERNETES-VERSION NAMESPACE SERVER
====    ================== ========= ======
default v1.7.0             kubeflow  https://35.184.102.122```
```

For this default environment, `env` will contain 
```
{
  "namespace": "kubeflow",
  "server": "https://35.184.102.122",
}
```

The main idea is that we should not have to define the `namespace` parameter per component, it will automatically be inherited by the environment in which we do a `ks apply <env>`